src/unix/aix.c

```diff
@@ -1097,13 +1102,26 @@ int uv_interface_addresses(uv_interface_address_t** addresses,
 
     address->name = uv__strdup(p->ifr_name);
 
-    if (p->ifr_addr.sa_family == AF_INET6) {
+    if (inet6) {
       address->address.address6 = *((struct sockaddr_in6*) &p->ifr_addr);
     } else {
       address->address.address4 = *((struct sockaddr_in*) &p->ifr_addr);
     }
 
-    /* TODO: Retrieve netmask using SIOCGIFNETMASK ioctl */
+    memset(address->phys_addr, 0, sizeof(address->phys_addr));
+    sa_addr = (struct sockaddr_dl*) &p->ifr_addr;
+    memcpy(address->phys_addr, LLADDR(sa_addr), sizeof(address->phys_addr));
+
+    if (ioctl(sockfd, SIOCGIFNETMASK, p) == -1) {
```

Comment:
There is no documented evidence about `ioctl` on `SIOCGIFNETMASK` could fail in any AIX versions due to known functional limitations but based on the runtime characteristics, the usual suspects apply:
`EAGAIN: Unable to allocate buffers for the output (p).`
`EBADF: Bad file descriptor (sockfd)`
`EINVAL: invalid pointer (in p)`
`EINTR: system call was interrupted`
...